bin/get-circle-artifact-url.js

```diff
@@ -0,0 +1,69 @@
+/** @format */
+
+const https = require( 'https' );
+
+const baseOptions = {
+	host: 'circleci.com',
+	port: 443,
+	headers: {
+		Accept: 'application/json',
+	},
+};
+
+const basePath = '/api/v1.1/project/github/Automattic/wp-calypso';
+
+async function getCircleArtifactUrl( pathMatchRegex ) {
+	if ( ! pathMatchRegex instanceof RegExp ) {
+		console.error( 'Expected a valid RegExp. Received: %o', pathMatchRegex );
+		process.exit( 1 );
+	}
+	try {
+		// Fetch recent successful master builds
+		const builds = await httpsGetJsonPromise( {
+			...baseOptions,
+			path: `${ basePath }/tree/master?filter=successful&limit=20`,
+		} );
+
+		const buildNumbersWithArtifacts = builds
+			.filter( build => build.has_artifacts )
+			.map( build => build.build_num );
+
+		for ( const buildNumber of buildNumbersWithArtifacts ) {
```

Comment:
TIL [`for...of`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of). Out of curiosity, why not [`forEach`](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach)?

Reply:
`forEach` would require `forEach( async () => {} )` in this case, which I believe would eagerly/concurrently call all of the provided async callbacks. We wouldn't check builds one by one, but would fire a bunch of network requests and could not deterministically get the most recent one.

Using the `for` loop lets us block the loop at the `await` so that the network requests are fired and processed one by one.